### PR TITLE
Fix NullPointerExceptions

### PR DESCRIPTION
**Pull request by triquell**
Fix NullPointerExceptions for when the Plugin has been installed but no new cloud "Docker Swarm" has been added and for when no "Docker Agent Template" has been added.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/docker/swarm/DeadAgentServiceReaperActor.java [hunk 1]
```diff
@@ -36,20 +36,22 @@ private void reapDeadAgentServices() throws IOException {
...
-            final Object result = new ListServicesRequest(dockerSwarmApiUrl, "label", "ROLE=jenkins-agent").execute();
-            for (ScheduledService service : (List<ScheduledService>) getResult(result, List.class)) {
-                Object tasks = new ListTasksRequest(dockerSwarmApiUrl, "service", service.Spec.Name).execute();
-                if (tasks != null) {
-                    for (Task task : (List<Task>) getResult(tasks, List.class)) {
-                        if (task.isComplete()) {
-                            LOGGER.info("Reaping service: " + service.Spec.Name);
-                            new DeleteServiceRequest(dockerSwarmApiUrl, service.Spec.Name).execute();
-                            break;
+            if ( DockerSwarmCloud.get() != null ) {
```
Comment (Roemer) [inline src/main/java/org/jenkinsci/plugins/docker/swarm/DeadAgentServiceReaperActor.java]: Please remove the spaces after / before the braces of the `if`

File: src/main/java/org/jenkinsci/plugins/docker/swarm/DockerSwarmCloud.java [hunk 2]
```diff
@@ -71,7 +71,9 @@ public DockerSwarmCloud(DockerServerEndpoint dockerHost, String dockerSwarmApiUr
         this.swarmNetwork = swarmNetwork;
         this.cacheDriverName = cacheDriverName;
         this.tunnel = tunnel;
-        this.agentTemplates = agentTemplates;
+        if ( agentTemplates != null ) {
```
Comment (Roemer) [inline src/main/java/org/jenkinsci/plugins/docker/swarm/DockerSwarmCloud.java]: Please remove the spaces after / before the braces of the `if`

File: src/main/java/org/jenkinsci/plugins/docker/swarm/OneShotProvisionQueueListener.java [hunk 3]
```diff
@@ -20,9 +20,11 @@
     @Override
     public void onEnterBuildable(final Queue.BuildableItem bi) {
         final Queue.Task job = bi.task;
-        final List<String> labels = DockerSwarmCloud.get().getLabels();
-        if (job.getAssignedLabel() != null && labels.contains(job.getAssignedLabel().getName())) {
-            BuildScheduler.scheduleBuild(bi);
+        if ( DockerSwarmCloud.get() != null ) {
```
Comment (Roemer) [inline src/main/java/org/jenkinsci/plugins/docker/swarm/OneShotProvisionQueueListener.java]: Please remove the spaces after / before the braces of the `if`

**Comment by Roemer** [comment]
Awesome, thanks for your contribution. That should completely fix #17 right?

**Comment by triquell** [comment]
> That should completely fix #17 right?

Sure! You can close it now :wink:

**Pull request merged**